packages/fluentui/react-bindings/src/hooks/useClasses.ts

```diff
@@ -0,0 +1,218 @@
+  variables?: ComponentVariablesInput;
+};
+
+export const useClasses = <StyleProps extends PrimitiveProps>(
+  displayName: string,
+  options: UseStylesOptions<StyleProps>,
+): string => {
+  const { disableAnimations, performance: performanceFlags, telemetry, theme, renderer, rtl } = useFluentContext();
+  const {
+    className: componentClassName = process.env.NODE_ENV === 'production' ? '' : 'no-classname-🙉',
+    mapPropsToStyles = () => ({} as StyleProps),
+    mapPropsToInlineStyles = () => ({} as InlineStyleProps<StyleProps>),
+  } = options;
+
+  const componentStylesProps = mapPropsToStyles();
+  const inlineStylesProps = mapPropsToInlineStyles();
+
+  //
+  // To compute styles we are going through three stages:
+  // - resolve variables (siteVariables => componentVariables + props.variables)
+  // - resolve styles (with resolvedVariables & props.styles & props.design)
+  // - compute classes (with resolvedStyles)
+  // - conditionally add sources for evaluating debug information to component
+
+  const telemetryVariablesStart = telemetry?.enabled ? performance.now() : 0;
+
+  let resolvedVariablesHash: string = '';
+  let resolvedVariables: ComponentVariablesObject = {};
+
+  if (typeof theme.componentVariables[displayName] === 'function') {
+    // @ts-ignore
+    if (theme.componentVariables[displayName].__hash) {
+      // @ts-ignore
+      resolvedVariables = theme.componentVariables[displayName].__resolved;
+      // @ts-ignore
+      resolvedVariablesHash = theme.componentVariables[displayName].__hash;
+    } else {
+      resolvedVariables = theme.componentVariables[displayName](theme.siteVariables);
+      resolvedVariablesHash = hash(JSON.stringify(resolvedVariables));
+
+      // @ts-ignore
+      theme.componentVariables[displayName].__resolved = resolvedVariables;
+      // @ts-ignore
+      theme.componentVariables[displayName].__hash = resolvedVariablesHash;
+    }
+  } else if (typeof theme.componentVariables[displayName] !== 'undefined') {
+    throw new Error();
+  }
+
+  if (telemetry?.enabled && telemetry.performance[displayName]) {
+    telemetry.performance[displayName].msResolveVariablesTotal += performance.now() - telemetryVariablesStart;
+  }
+
+  const { className, design, styles, variables } = inlineStylesProps;
+  const noInlineStylesOverrides = !(design || styles);
+
+  let noVariableOverrides = performanceFlags.enableBooleanVariablesCaching || !variables;
+
+  /* istanbul ignore else */
+  if (process.env.NODE_ENV !== 'production') {
+    if (!performanceFlags.enableStylesCaching && performanceFlags.enableBooleanVariablesCaching) {
+      throw new Error(
+        '@fluentui/react-northstar: Please check your "performance" settings on "Provider", to enable "enableBooleanVariablesCaching" you need to enable "enableStylesCaching"',
+      );
+    }
+  }
+
+  if (performanceFlags.enableBooleanVariablesCaching) {
```

Comment:
nit: declare the variable closer to the usage

```suggestion
  let noVariableOverrides = performanceFlags.enableBooleanVariablesCaching || !variables;
  if (performanceFlags.enableBooleanVariablesCaching) {
```